src/utils.ts

```diff
@@ -371,9 +371,10 @@ export function uuidV4(): Buffer {
  */
 export function maxWireVersion(topologyOrServer?: Connection | Topology | Server): number {
   if (topologyOrServer) {
-    if (topologyOrServer.loadBalanced) {
-      // Since we do not have a monitor, we assume the load balanced server is always
-      // pointed at the latest mongodb version. There is a risk that for on-prem
+    if (topologyOrServer.loadBalanced || topologyOrServer.serverApi?.version) {
+      // Since we do not have a monitor, we assume
```

Comment:
Can we update this comment to reflect the actual implementation? We don't have a monitor when in load balanced mode but we do have a monitor when specifying `serverApi` and not in load balanced mode.